### DISCUSSION TITLE
🕳 Filling in Blackhole: Identifying Unsustainable Processes with SD

### DISCUSSION THREAD

**Discussion by hyunjimoon** [Data4DM/BayesSD#52] [people; relating]
## Stock-flow consistent model
<img width="856" alt="image" src="https://user-images.githubusercontent.com/30194633/194778574-8c03a4de-1e28-4ee5-a5d7-26899a760e14.png">

- balance sheets are updated in any period, SFC models can be used to identify unsustainable processes, for example a prolonged deficit of a sector will result in an unsustainable stock of debt

- consistent accounting framework prevents the modellers from leaving "black holes" i.e., unexplained parts of the model.

From [Seven unsustainable process](https://www.levyinstitute.org/pubs/sa/sevenproc.pdf), Wynne Godley identified seven unsustainable processes in US below with the above Stock-flow consistent:
(1) the fall in private saving into ever deeper negative territory, 
(2) the rise in the flow of net lending to the private sector, 
(3) the rise in the growth rate of the real money stock, 
(4) the rise in asset prices at a rate that far exceeds the growth of profits (or of GDP), 
(5) the rise in the budget surplus, 
(6) the rise in the current account deficit, 
(7) the increase in the United States's net foreign indebtedness relative to GDP.

**Comment by hyunjimoon** [comment]
Q1. Which direction between disaggregation (compartment to agent-based) vs aggregation (agent-based to compartment) do you think is easier for testing indirect structure? Designing built-in internal consistency checks (e.g. Hazhir commented on Ali's presentation saying certain parameter vectors specified from sensitivity test are unreasonable) is hard but is important. If you were given the choice of model development direction, which would you choose between branching (disaggregating) vs merging (aggregating) and why? From Yaman's talk:

<img width="1090" alt="image" src="https://user-images.githubusercontent.com/30194633/194835594-36df9670-0ce1-494a-a8af-c57e9ae9978e.png">

Yaman's insightful comment from LTA talk (attached below): "The totality of relations (in real system or model) is called the structure (of real system or model)."

Q2. This is relevant to our discussion on #32 and #34 in that hierarchical Bayesian model is middle-out approach starting from between compartment and agent-based modeling and can go either direction. Could we borrow mathematical formulation for middle-out approach in hierarchical time-series from below? Details [here](https://otexts.com/fpp3/single-level.html#middle-out-approach).

<img width="940" alt="image" src="https://user-images.githubusercontent.com/30194633/194837246-50bb964e-e827-4ddc-bbd4-6a40256faaf4.png">

Q3. Do we need internal consistency check along disaggregation? Or do we need to frame this as tradeoff between computation cost and accuracy? To be specific, is two compartment model always a special case of three compartment model? What internal consistency check do we additionally need when extending from S-I-R to S-E-I-R model?

Tagging who might be able to help me out @zhasgul, @NzL94, @tomfid @jandraor @Dashadower 

Ref.
[Barlas22_LTATalk.pdf](https://github.com/Data4DM/BayesSD/files/9744622/Barlas22_LTATalk.pdf)
[Gonenc21_SESDYN_OutputAnalysis.pdf](https://github.com/Data4DM/BayesSD/files/9744623/Gonenc21_SESDYN_OutputAnalysis.pdf)

**Reply by NzL94** [reply]
Q1: I am tempted to say we want things to be as asggregated (or simplified) as possible at the beginning. I would definitely try to come up with some modeling assumptions that could help me make the most of the data. After that, I would start to think of which assumptions can be relaxed and why in order to capture deviations from the general model I have devised. This is no doubt a consequence of my background in Economics, but I believe this is the best way to approach the problem.

Q2: I believe this is doable provided we know that hierarchies are part of the problem, that is, as long as we know that the data that we want to fit to the model comes in groups (not that strong of an assumption)

Q3: Ok, here we need to be very precise. SIR is the general case of the SEIR, so in this case, since they both should generate the same limiting behavior (i.e. either pandemic or disease dying out), we could probably frame this as a trade-off between computation and accuracy. But we cannot make the same assumption about the SI model: it generates a completely different steady state! And this is where the internal consistency is important and should be coming form our modeling choices.

**Reply by tomfid** [reply]
Q1: I tend to agree with NzL94, primarily because working with disaggregated data can be really expensive (not just computation - collection and interpretation time may be the bigger part). If your model has broad scope, you're likely to be better off with shallow detail in any one area; otherwise you'll never finish.

Q2: I think things are rarely as easy as the easy change of levels implied by the grouped & middle-out options. For example, in the CWD model, disaggregating by deer age & sex implies major structural changes, new dynamics, and different interpretation of some parameters.

Q3: I think there are some very basic checks one might want, like verifying that the system is conserving people and dimensionally consistent. Then I guess I'd be interested in whether the SEIR behavior approaches the SIR behavior as tauE becomes small. SEIR might not be the greatest example, because it's well understood, so you're not likely to get into trouble.
